### ISSUE TITLE
[V12] Leaves Accrual doesn't Work

### ISSUE THREAD

**Issue by refana77** [odoo/odoo#32100]
Impacted versions: Odoo V12

Steps to reproduce:
- Goto employees --> Create an employee
- Goto Leaves -- > Configuration --> Leave Types, create a leave type  'Annual Leaves' and the leave validity start date is from 1st jan 2019 (please see the screenshot attached)

-From Leaves create a new allocation request with mode 'By Employee' and select the employee name. activate the accrual and give the condition:
Add1.00Day(s)of leaves every1Week(s)

(please see the screenshot attached)

Current behavior:
After running the scheduler 'Accrual Leave', no days are allocated to the employee

Expected behavior:
If we activated the accrual option, based on the accrual condition given on the allocation request the days should be allocated (11 days as per today's date 25 March)
 
As you can see from the screenshot attached, the validity for the Annual leaves starts from 1 Jan 2019, if I applied an allocation request with accrual using this condition I should get 11 days to leave allocation for that employee irrespective of employee creation date. 

Video/Screenshot link (optional):
![Leave type](https://user-images.githubusercontent.com/43813739/54908974-2b7e5180-4efa-11e9-843c-8bb25f476334.png)
![leave allocation](https://user-images.githubusercontent.com/43813739/54908990-32a55f80-4efa-11e9-99bd-8dbf1a428f4f.png)
![Leaves alllocated after approving leaves](https://user-images.githubusercontent.com/43813739/54908995-35a05000-4efa-11e9-8eea-66b7595dc47c.png)

**Comment by sswapnesh** [comment]
Hello @refana77 
> If we activated the accrual option, based on the accrual condition given on the allocation request the days should be allocated (11 days as per today's date 25 March)

Allocation will be started from next week in this case. (Eg, Next leave will be allocated on 1st April)

> the validity for the Annual leaves starts from 1 Jan 2019, if I applied an allocation request with accrual using this condition I should get 11 days to leave allocation for that employee irrespective of employee creation date.

`Annual leaves starts from 1 Jan 2019` It is for the Leaves validity, So employee can use this leave type out of Start/End date specified in configuration.

**Comment by pedrobaeza** [comment]
@sswapnesh, then no valid issue in your opinion?

**Comment by sswapnesh** [comment]
IMO, This is desired behavior. 
(This can be closed)

**Comment by pedrobaeza** [comment]
@sswapnesh I see your reasoning, but somehow can induce to confusion, so I don't know exactly what to do.

@alexey-pelykh what do you think about this?

**Comment by alexey-pelykh** [comment]
@pedrobaeza @sswapnesh as you know I find current accrual leave allocation in Odoo very confusing and partially improper, thus authored this module https://github.com/OCA/hr/pull/501 (`hr_holidays_accrual`) that provides better experience and much better control over the how accruement is made.

Especially that creation date of the employee entry is taken into account:
https://github.com/odoo/odoo/blob/9ea4fefed2f94c6d211951e573b9a9a456f00ce6/addons/hr_holidays/models/hr_leave_allocation.py#L170

and

https://github.com/odoo/odoo/blob/9ea4fefed2f94c6d211951e573b9a9a456f00ce6/addons/hr_holidays/models/hr_leave_allocation.py#L177

**Comment by pedrobaeza** [comment]
Let's tag this for getting Odoo's opinion

**Comment by alexey-pelykh** [comment]
@refana77 with the module mentioned above this would look like:
![image](https://user-images.githubusercontent.com/345102/54975661-7c488580-4fa0-11e9-9712-8a01e35a5665.png)

**Comment by refana77** [comment]
If it is based on employee creation date then what's the point if the company start to use the system at the mid of the year and the leave type validity set from the beginning of the year. Accrual concept will not work, right?

**Comment by refana77** [comment]
> @refana77 with the module mentioned above this would look like:
> ![image](https://user-images.githubusercontent.com/345102/54975661-7c488580-4fa0-11e9-9712-8a01e35a5665.png)

from where you got this screenshot. Is it from any custom module? sorry I didn't find this form on the system

**Comment by alexey-pelykh** [comment]
> If it is based on employee creation date then what's the point if the company start to use the system at the mid of the year and the leave type validity set from the beginning of the year. Accrual concept will not work, right?

My point exactly, you'll have to edit that field manually via database or other hidden/for-administrators UI.

**Comment by alexey-pelykh** [comment]
> Is it from any custom module?

@refana77 Yes, this is a custom module (https://github.com/OCA/hr/pull/501), to be included in OCA (once @pedrobaeza would help me out with that) and should be available on the App Store soon after that.

**Comment by refana77** [comment]
@alexey-pelykh thank you for your valuable contribution

Is there any way to get the accrual working in the system properly without using any customization.

**Comment by alexey-pelykh** [comment]
@refana77 none I'm aware of, and none I'd adopt using

**Comment by sswapnesh** [comment]
Hello @tivisse @RomainLibert ,
Could please add your thoughts on this if this is expected behaviour or not ?

**Comment by RomainLibert** [comment]
https://github.com/odoo/odoo/commit/90731ad170c503cdfe89a9998fa1d1e2a5035c86#diff-4b74c5cd47565e7fbe2a526d12a577c4
This should partially fix that problem, it is already merged in 12.0. 
There is probably little more that we can do to fix this in a stable release.

**Comment by refana77** [comment]
Can anyone please explain is it an expected behavior or not

**Comment by 3mrdev** [comment]
@refana77 Did you check the "Scheduled Actions/Accrual Time Off: Updates the number of time off" ?
